packages/app-builder-lib/src/targets/MsiWrappedTarget.ts

```diff
@@ -0,0 +1,224 @@
+import { Arch, log, deepAssign } from "builder-util"
+import { UUID } from "builder-util-runtime"
+import { getBinFromUrl } from "../binDownload"
+import * as ejs from "ejs"
+import { readFile, writeFile } from "fs/promises"
+import { Lazy } from "lazy-val"
+import * as path from "path"
+import { MsiWrappedOptions } from "../"
+import { Target, TargetConfiguration } from "../core"
+import { FinalCommonWindowsInstallerOptions, getEffectiveOptions } from "../options/CommonWindowsInstallerConfiguration"
+import { getTemplatePath } from "../util/pathManager"
+import { VmManager } from "../vm/vm"
+import { WineVmManager } from "../vm/WineVm"
+import { WinPackager } from "../winPackager"
+import { createStageDir } from "./targetUtil"
+
+const ELECTRON_BUILDER_UPGRADE_CODE_NS_UUID = UUID.parse("44a7d685-ff0b-4877-b761-5dc194e7f071")
+const ELECTRON_MSI_WRAPPED_NS_UUID = UUID.parse("467f7bb2-a83c-442f-b776-394d316e8e53")
+
+const projectTemplate = new Lazy<(data: any) => string>(async () => {
+  const template = (await readFile(path.join(getTemplatePath("msiWrapped"), "template.xml"), "utf8"))
+    .replace(/{{/g, "<%")
+    .replace(/}}/g, "%>")
+    .replace(/\${([^}]+)}/g, "<%=$1%>")
+  return ejs.compile(template)
+})
+
+// WiX doesn't support Mono, so, dontnet462 is required to be installed for wine (preinstalled in our bundled wine)
+export default class MsiWrappedTarget extends Target {
```

Comment:
I'm noticing a **lot** of copied code from `MsiTarget`, perhaps we should extend `MsiTarget` instead of `Target`?
This would mirror the current setup of 
```
interface MsiWrappedOptions extends CommonWindowsInstallerConfiguration, TargetSpecificOptions { ... }
```

Reply:
Good point, I copied it at the start not knowing how far it may stray, let me try to clean that up